agent/engine/docker_task_engine.go

```diff
@@ -392,7 +395,9 @@ func (engine *DockerTaskEngine) CheckTaskState(task *api.Task) {
 		if !ok {
 			continue
 		}
-		status, metadata := engine.client.DescribeContainer(dockerContainer.DockerID)
+		ctx, cancel := context.WithCancel(engine.ctx)
```

Comment:
Same question as above. Can't you just use `engine.ctx` directly?

Reply:
See comments above.